### PR TITLE
fix: nested or nil mutation processing

### LINKED ISSUE #492: testnet mutation error
```
** (Mix) Could not start application ae_mdw: bad return value: {{AeMdw.Application, :start_phase, [:migrate_db, :normal, []]}, {:EXIT, {{:badmatch, {:aborted, {%Protocol.UndefinedError{description: "", protocol: AeMdw.Db.Mutation, value: []}, [{AeMdw.Db.Mutation, :impl_for!, 1, [file: 'lib/ae_mdw/db/mutation.ex', line: 1]}, {AeMdw.Db.Mutation, :mutate, 1, [file: 'lib/ae_mdw/db/mutation.ex', line: 16]}, {Enum, :"-each/2-lists^foreach/1-0-", 2, [file: 'lib/enum.ex', line: 783]}, {Enum, :each, 2, [file: 'lib/enum.ex', line: 783]}, {:mnesia_tm, :apply_fun, 3, [file: 'mnesia_tm.erl', line: 842]}, {:mnesia_tm, :execute_transaction, 5, [file: 'mnesia_tm.erl', line: 818]}, {AeMdw.Mnesia, :transaction, 1, [file: 'lib/ae_mdw/mnesia.ex', line: 161]}, {AeMdw.Migrations.OracleRespondInternalCalls, :run, 1, [file: 'priv/migrations/20220124162300_oracle_respond_internal_calls.ex', line: 40]}, {Mix.Tasks.MigrateDb, :apply_migration!, 2, [file: 'lib/mix/tasks/migrate_db.ex', line: 87]}, {Enum, :"-map/2-lists^map/1-0-", 2, [file: 'lib/enum.ex', line: 1396]}, {Mix.Tasks.MigrateDb, :run, 1, [file: 'lib/mix/tasks/migrate_db.ex', line: 40]}, {AeMdw.Application, :start_phase, 3, [file: 'lib/ae_mdw/application.ex', line: 238]}, {:application_starter, :run_the_phase, 4, [file: 'application_starter.erl', line: 99]}, {:application_starter, :start_apps, 3, [file: 'application_starter.erl', line: 59]}, {:application_starter, :start, 3, [file: 'application_starter.erl', line: 45]}, {:application_master, :start_the_app, 5, [file: 'application_master.erl', line: 316]}, {:application_master, :start_it_new, 7, [file: 'application_master.erl', line: 299]}]}}}, [{AeMdw.Mnesia, :transaction, 1, [file: 'lib/ae_mdw/mnesia.ex', line: 160]}, {AeMdw.Migrations.OracleRespondInternalCalls, :run, 1, [file: 'priv/migrations/20220124162300_oracle_respond_internal_calls.ex', line: 40]}, {Mix.Tasks.MigrateDb, :apply_migration!, 2, [file: 'lib/mix/tasks/migrate_db.ex', line: 87]}, {Enum, :"-map/2-lists^map/1-0-", 2, [file: 'lib/enum.ex', line: 1396]}, {Mix.Tasks.MigrateDb, :run, 1, [file: 'lib/mix/tasks/migrate_db.ex', line: 40]}, {AeMdw.Application, :start_phase, 3, [file: 'lib/ae_mdw/application.ex', line: 238]}, {:application_starter, :run_the_phase, 4, [file: 'application_starter.erl', line: 99]}, {:application_starter, :start_apps, 3, [file: 'application_starter.erl', line: 59]}, {:application_starter, :start, 3, [file: 'application_starter.erl', line: 45]}, {:application_master, :start_the_app, 5, [file: 'application_master.erl', line: 316]}, {:application_master, :start_it_new, 7, [file: 'application_master.erl', line: 299]}]}}}
make: *** [Makefile:37: shell] Segmentation fault (core dumped)
```

Reproducible with testnet .art database snapshot.

### PR DESCRIPTION
**Pull request by jyeshe**
Fixes #492

### CODE REVIEW

**Comment by jyeshe** [comment]
fixes for resuming sync with a snapshot but another issue happens when beginning from scratch

**Comment by jyeshe** [comment]
> fixes for resuming sync with a snapshot but another issue happens when beginning from scratch

pre-existing issue with v1.4.0 on:
`      {:tx, _, _, {mdw_height, _}, _} = read_tx!(mdw_tx_index)`

the mdw tracks sync height by transactions which start to happen only at height 5133

File: lib/ae_mdw/db/sync/transaction.ex [hunk 1]
```diff
@@ -224,12 +223,11 @@ defmodule AeMdw.Db.Sync.Transaction do
...
-      List.flatten([
-        MnesiaWriteMutation.new(Model.Block, mb_model),
-        txs_mutations,
-        Aex9AccountPresenceMutation.new(height, mbi)
-      ])
+    mutations = [
+      MnesiaWriteMutation.new(Model.Block, mb_model),
+      txs_mutations,
+      Aex9AccountPresenceMutation.new(height, mbi)
+    ]
```
Comment (sborrazas) [inline lib/ae_mdw/db/sync/transaction.ex:230]: This is going to simplify building mutations in several places, good idea

File: lib/ae_mdw/mnesia.ex [hunk 2]
```diff
@@ -159,7 +159,10 @@ defmodule AeMdw.Mnesia do
   def transaction(mutations) do
     {:atomic, :ok} =
       :mnesia.sync_transaction(fn ->
-        Enum.each(mutations, &Mutation.mutate(&1))
+        mutations
+        |> List.flatten()
+        |> Enum.reject(&is_nil/1)
+        |> Enum.each(&Mutation.mutate/1)
```
Comment (sborrazas) [inline lib/ae_mdw/mnesia.ex:165]: Perhaps we can perform all of these computations outside of the transaction?

Comment (jyeshe) [reply lib/ae_mdw/mnesia.ex]: done

**Review by sborrazas — APPROVED** [review]

**Review by thepiwo — APPROVED** [review]

**Pull request merged**